Tween/Serialization/TweenBuilder.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+
+namespace DUCK.Tween.Serialization
+{
+	public class TweenBuilder : MonoBehaviour
+	{
+		private const string NULL_COMPONENT_ERROR = "Cannot build tween for {0}. Argument {1} ({2}) was not serialized and could not find the component on the TweenBuilder";
+
+		[SerializeField]
+		private bool playOnStart;
+
+		[SerializeField]
+		private bool useReferencedConfig;
+
+		[SerializeField]
+		private TweenConfigScriptableObject tweenConfigScriptableObject;
+
+		[SerializeField]
+		private TweenConfig tweenConfig = new TweenConfig();
+		public TweenConfig TweenConfig
+		{
+			get { return tweenConfig; }
+			set { tweenConfig = value; }
+		}
+
+		private void Start()
+		{
+			if (playOnStart)
+			{
+				Play();
+			}
+		}
+
+		public void Play()
+		{
+			var config = useReferencedConfig ? tweenConfigScriptableObject.Config : TweenConfig;
+			var tween = BuildTween(config, gameObject);
+			tween.Play();
+		}
+
+#if UNITY_EDITOR
+		public bool PlayOnStart
```

Comment:
Why should this not be accessible outside editor time?

Reply:
No i wouldn't have but wanted to understand where it was being used but if only on a inspector class then I don't mind.